### PR TITLE
Add Skeletons and Blend Shapes to the OpenGL renderer

### PR DESCRIPTION
**Pull request by clayjohn**
This uses a similar multipass approach to blend shapes as Godot 3.x, the major difference here is that we need to convert the normals and tangents to octahedral for rendering.

Skeletons work the same as the Vulkan renderer except the bones are stored in a texture as they were in 3.x.

Performance in my testing is very good on newer hardware. I expect performance will be not amazing on older hardware and on Apple devices due to the use of transform feedback. The plan is to use ANGLE to improve performance on Windows and on Apple devices. 

This is the last big 2D feature remaining for the OpenGL renderer. Hooray!

### CODE REVIEW

**Comment by fire** [comment]
image

[female.zip](https://github.com/godotengine/godot/files/10109956/female.zip)

I tried loading this and it failed. Can you check?

**Comment by fire** [comment]
image

Tested with the web platform and got a slate grey screen.

File: drivers/gles3/shaders/skeleton.glsl [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	// Binormal sign encoded in y component
+	vec2 oct = vec2(oct_sign_encoded.x, abs(oct_sign_encoded.y) * 2.0 - 1.0);
+	return vec4(oct_to_vec3(oct), sign(oct_sign_encoded.y));
+}
+
+vec2 tang_to_oct(vec4 base) {
+	vec2 oct = vec3_to_oct(base.xyz);
+	// Encode binormal sign in y component
+	oct.y = oct.y * 0.5f + 0.5f;
+	oct.y = base.w >= 0.0f ? oct.y : 1 - oct.y;
```
Comment (fire) [inline drivers/gles3/shaders/skeleton.glsl:119]:
```suggestion
	oct.y = base.w >= 0.0f ? oct.y : 1.0 - oct.y;
```

Comment (fire) [reply drivers/gles3/shaders/skeleton.glsl:119]: This fixes WebGL2

Comment (clayjohn) [reply drivers/gles3/shaders/skeleton.glsl]: Fixed! Thanks

**Review by fire — APPROVED** [review]
The feature works on opengl and on webgl2. I had some trouble with the colours of the outputs, but unclear what code is responsible. Mesh normals?

I did black box testing and not code review.

It is an improvement and we can work on the bugs on a different pr.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**